sched/scheduler/stateful_scheduler.go

```diff
@@ -412,3 +434,61 @@ func (s *statefulScheduler) scheduleTasks() {
 			})
 	}
 }
+
+/**
+Put the kill request on channel that is processed by the main
+scheduler loop, and wait for the response
+*/
+func (s *statefulScheduler) KillJob(jobId string) error {
+
+	responseCh := make(chan error, 1)
+	req := jobKillRequest{jobId: jobId, responeCh: responseCh}
+	s.killJobCh <- req
+
+	return <-req.responeCh
+
+}
+
+// process all requests verifying that the jobIds exist:  Send errors back
+// immediately on the request channel for jobId that don't exist, then
+// kill all the jobs with a valid ID
+//
+// this function is part of the main scheduler loop
+func (s *statefulScheduler) killJobs() {
+	var validKillRequests []jobKillRequest
+
+	// validate jobids and sending invalid ids back and building a list of valid ids
+	for haveKillRequest := true; haveKillRequest == true; {
+		select {
+		case req := <-s.killJobCh:
+			// can we find the job?
+			_, ok := s.inProgressJobs[req.jobId]
+			if !ok {
+				req.responeCh <- fmt.Errorf("Job Id %s, not found. "+
+					" The job may be finished, "+
+					" the request may still be in the queue to be scheduled, or "+
+					" the id may be invalid.  "+
+					" Check the job status, verify the id and/or resubmit the kill request after a few moments.",
+					req.jobId)
+			} else {
+				validKillRequests = append(validKillRequests[:], req)
+			}
+		default:
+			haveKillRequest = false
+		}
+	}
+
+	// kill the jobs with valid ids
+	for _, req := range validKillRequests {
+		jobState := s.inProgressJobs[req.jobId]
+		for _, task := range jobState.Tasks {
+			if task.Status != sched.Completed {
+				if task.TaskRunner != nil {
+					task.TaskRunner.abortCh <- 1
+				}
+			}
+		}
+
```

Comment:
added jobKilled back in.

Have to talk with you about closing abortCh ... this is where we trigger the abort action, if we close it then how does taskRunner.abortRequested() ever become true?